Feature/custom rule example (#67)
AbsaOSS/enceladus: examples/README.md

                    Copyright 2018 ABSA Group Limited
                  
      Licensed under the Apache License, Version 2.0 (the "License");
      you may not use this file except in compliance with the License.
                You may obtain a copy of the License at
               http://www.apache.org/licenses/LICENSE-2.0
            
     Unless required by applicable law or agreed to in writing, software
       distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      See the License for the specific language governing permissions and
                      limitations under the License.

# Examples of Enceladus usage

> ## Custom Rules Implementations and usage

Reviewer: ```suggestion
## Custom Rule implementations and usage
```

> ## Custom Rules Implementations and usage

Reviewer: Would also be good to mention that example test cases can be found at: 
[/examples/src/test/scala/src/main/scala/za/co/absa/enceladus/examples/interpreter/rules/custom](https://github.com/AbsaOSS/enceladus/tree/feature/custom_rule_example/examples/src/test/scala/src/main/scala/za/co/absa/enceladus/examples/interpreter/rules/custom)
which has an extra "src/main/scala/" in there and really should be:
/examples/src/test/scala/za/co/absa/enceladus/examples/interpreter/rules/custom

Author: Both addressing, and moving also the examples themselves to correct folder

### CustomRuleSample1

> Very simple example how showing how a custom rule is declared, what are its very basic parts and how it transforms some data.

Reviewer: ```suggestion
Very simple example showing how a custom rule is declared, what its very basic parts are and how it transforms hardcoded data.
```
I would also describe the purpose of the rule, what transformation it performs.

### CustomRuleSample2

> Somewhat advanced example showing two possible implementations of a LPAR/RPAD custom rule. The example includes hierarchy

Reviewer: ```suggestion
Somewhat advanced example showing two possible implementations of a LPAR/RPAD custom rule. The example includes a hierarchy
```

> of classes and usage of the new rules aon soe chunk of dat again. 

Reviewer: ```suggestion
of classes and usage of the new rules on hardcoded data. 
```

### CustomRuleSample3

> Using the previously implemented custom rules an actual dataset is loaded from a CSV file, transformed and the result is shown 

Reviewer: ```suggestion
Using the previously implemented custom LPAD/RPAD rules an actual dataset is loaded from a CSV file (`examples/data/input/example_data.csv`), transformed and the result is shown.
```

### CustomRuleSample4
Command line driven example. Data are loaded from the file specified (supporting multiple formats), transformed and 
saved to a csv file on a specified path.  
NB! As the transformation rules are hardcoded the input file have to have a column named **text_column**

For command-line arguments description run:  
`spark-submit --class src.main.scala.za.co.absa.enceladus.examples.CustomRuleSample4 --master local enceladus-examples.jar --help`

Example arguments:  

> `spark-submit --class src.main.scala.za.co.absa.enceladus.examples.CustomRuleSample4 --master local enceladus-examples.jar --input-file="data/input/example_data.csv" --out-path="data/output" --header=1`

Reviewer: ```suggestion
`spark-submit --class src.main.scala.za.co.absa.enceladus.examples.CustomRuleSample4 --master local enceladus-examples.jar --input-file="data/input/example_data.csv" --out-path="data/output" --header=true`
```
I think that makes it clearer

Author: Yup

# Examples of Enceladus usage
## Custom Rules Implementations and usage
### Requirements

> * Spark 2.4+ (might work with earlier, not tested)

Reviewer: Definitely works with 2.2.1, which is the spark we are using in our environments. Also, I tested your code on my machine running 2.2.1.
